bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/handler/TiVoHandler.java

```diff
@@ -0,0 +1,364 @@
+        this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                "Power on device or check network configuration/connection.");
+    }
+
+    private void sendCommand(String commandKeyword, String commandParameter, TivoStatusData currentStatus) {
+        if (!tivoConnection.isPresent()) {
+            return;
+        }
+
+        TivoStatusData deviceStatus = tivoConnection.get().getServiceStatus();
+        TivoStatusData commandResult = null;
+        logger.debug("handleCommand '{}' - {} found!", getThing().getUID(), commandKeyword);
+        // Re-write command keyword if we are in STANDBY, as only IRCODE TIVO will wake the unit from
+        // standby mode
+        if (deviceStatus.getConnectionStatus() == ConnectionStatus.STANDBY && commandKeyword.contentEquals("TELEPORT")
+                && commandParameter.contentEquals("TIVO")) {
+            commandKeyword = "IRCODE " + commandParameter;
+            logger.debug("TiVo '{}' TELEPORT re-mapped to IRCODE as we are in standby: '{}'", getThing().getUID(),
+                    commandKeyword);
+        }
+        // Execute command
+        if (commandKeyword.contentEquals("FORCECH") || commandKeyword.contentEquals("SETCH")) {
+            commandResult = chChannelChange(commandKeyword, commandParameter);
+        } else {
+            commandResult = tivoConnection.get().cmdTivoSend(commandKeyword + " " + commandParameter);
+        }
+
+        // Post processing
+        if (commandResult != null && commandParameter.contentEquals("STANDBY")) {
+            // Force thing state into STANDBY as this command does not return a status when executed
+            commandResult.setConnectionStatus(ConnectionStatus.STANDBY);
+        }
+
+        // Push status updates
+        if (commandResult != null && commandResult.isCmdOk()) {
+            updateTivoStatus(currentStatus, commandResult);
+        }
+
+        // disconnect once command is complete (really only disconnects if isCfgKeepConnOpen = false)
+        tivoConnection.get().connTivoDisconnect(false);
+    }
+
+    int convertValueToInt(Object value) {
+        if (value instanceof BigDecimal) {
+            return ((BigDecimal) value).intValue();
+        }
+        if (value instanceof String) {
+            return Integer.valueOf((String) value);
+        }
+        if (value instanceof Double) {
+            return ((Double) value).intValue();
+        }
+        return (Integer) value;
+    }
+
+    boolean convertValueToBoolean(Object value) {
+        return value instanceof Boolean ? ((Boolean) value) : Boolean.valueOf((String) value);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing a TiVo '{}' with config options", getThing().getUID());
+
+        tivoConfigData = getConfigAs(TivoConfigData.class);
+
+        tivoConfigData.setCfgIdentifier(getThing().getUID().getAsString());
+        logger.debug("TivoConfigData Obj: '{}'", tivoConfigData);
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging